inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp

```diff
@@ -812,575 +813,602 @@ void MKLDNNNormalizeL2Node::setPostOps(mkldnn::primitive_attr &attr, bool initWe
+}
+
+void MKLDNNNormalizeL2Node::execute(mkldnn::stream strm) {
+    if (!execPtr)
+        THROW_ERROR << "doesn't have a compiled executor.";
+
+    const uint8_t *src_ptr = reinterpret_cast<const uint8_t *>(getParentEdgeAt(DATA)->getMemoryPtr()->GetPtr());
+    uint8_t *dst_ptr = reinterpret_cast<uint8_t *>(getChildEdgeAt(DATA)->getMemoryPtr()->GetPtr());
+    execPtr->exec(src_ptr, dst_ptr);
+}
+
+std::vector<VectorDims> MKLDNNNormalizeL2Node::shapeInfer() const {
+    return std::vector<VectorDims>{getParentEdgesAtPort(DATA)[0]->getMemory().getStaticDims()};
+}
+
+// *====================* CornerCase *===================*
+
+template <typename in_data_t, typename out_data_t>
+class MKLDNNNormalizeL2Node::NormalizeL2CornerCaseExecutor : public MKLDNNNormalizeL2Node::NormalizeL2Executor {
+public:
+    NormalizeL2CornerCaseExecutor(const VectorDims& dims) {
+        workAmount = std::accumulate(dims.begin(), dims.end(), 1, std::multiplies<size_t>());
+    }
+
+    void exec(const uint8_t *src_ptr, uint8_t *dst_ptr) override {
+        normalize(reinterpret_cast<const in_data_t*>(src_ptr), reinterpret_cast<out_data_t*>(dst_ptr));
+    }
+private:
+    void normalize(const in_data_t* src_data, out_data_t* dst_data) {
+        parallel_for(workAmount, [&](size_t i) {
+            dst_data[i] = src_data[i] == 0 ? 0 : 1;
+        });
+    }
+
+    size_t workAmount = 0lu;
+};
+
+// *=================* *======* *=================*
+
+// *=================* JIT case *=================*
+
+template <typename in_data_t, typename out_data_t>
+class MKLDNNNormalizeL2Node::NormalizeL2JitExecutor : public MKLDNNNormalizeL2Node::NormalizeL2Executor {
+public:
+    NormalizeL2JitExecutor(const NormalizeL2Attrs& attrs_, const mkldnn::primitive_attr& kernel_attrs, const VectorDims& dims) : attrs(attrs_) {
+        if (!attrs.is_nchw && !attrs.is_nhwc && !attrs.is_blk) {
+            IE_THROW() << "Normalaize2L executor has selected layout which is not supported";
         }
 
-        jcp.across_spatial = across_spatial;
-        auto dims = getParentEdgeAt(0)->getMemory().getStaticDims();
+        jcp.src_dt = MKLDNNExtensionUtils::IEPrecisionToDataType(attrs.input_prec);
+        jcp.dst_dt = MKLDNNExtensionUtils::IEPrecisionToDataType(attrs.output_prec);
+        jcp.src_data_size = attrs.input_prec.size();
+        jcp.dst_data_size = attrs.output_prec.size();
+        jcp.across_spatial = attrs.across_spatial;
+
+        jcp.is_nchw = attrs.is_nchw;
+        jcp.is_nhwc = attrs.is_nhwc;
+        jcp.is_blk = attrs.is_blk;
+
         size_t dims_size = dims.size();
-        jcp.n = (dims_size > 0) ? dims[0] : 1lu;
-        jcp.c = (dims_size > 1) ? dims[1] : 1lu;
+        jcp.n = dims[0];
+        jcp.c = dims[1];
         jcp.h = (dims_size > 2) ? dims[2] : 1lu;
         jcp.w = (dims_size > 3) ? dims[3] : 1lu;
 
         if (mayiuse(cpu::x64::avx512_common)) {
+            attrs.blk_size = 16;
```

Comment:
minor
looks like we use and initialize `blk_size` only inside `NormalizeL2JitExecutor`
so maybe better to make it class field and remove from `NormalizeL2Attrs`?

Reply:
Agree, it would be better. Done, Thanks!